delivery_postlogistics/models/stock_picking.py

```diff
@@ -0,0 +1,259 @@
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import base64
+from operator import attrgetter
+
+import lxml.html
+
+from odoo import _, api, exceptions, fields, models
+
+from ..postlogistics.web_service import PostlogisticsWebService
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    delivery_fixed_date = fields.Date(
+        "Fixed delivery date", help="Specific delivery date (ZAW3217)"
+    )
+
+    # TODO: consider refactoring these fields using a partner relation instead
+    delivery_place = fields.Char(help="For Deposit item service (ZAW3219)")
+    delivery_phone = fields.Char(
+        "Phone", help="For notify delivery by telephone (ZAW3213)"
+    )
+    delivery_mobile = fields.Char(
+        "Mobile", help="For notify delivery by telephone (ZAW3213)"
+    )
+
+    def _get_packages_from_picking(self):
+        """Get all the packages from the picking"""
+        self.ensure_one()
+        operation_obj = self.env["stock.move.line"]
+        operations = operation_obj.search(
+            [
+                "|",
+                ("package_id", "!=", False),
+                ("result_package_id", "!=", False),
+                ("picking_id", "=", self.id),
+            ]
+        )
+        package_ids = set()
+        for operation in operations:
+            # Take the destination package. If empty, the package is
+            # moved so take the source one.
+            package_ids.add(operation.result_package_id.id or operation.package_id.id)
+
+        packages = self.env["stock.quant.package"].browse(package_ids)
+        return packages
+
+    def get_shipping_label_values(self, label):
+        # TODO: consider to depends on base_delivery_carrier_label
+        self.ensure_one()
+        return {
+            "name": label["name"],
+            "res_id": self.id,
+            "res_model": "stock.picking",
+            "datas": label["file"],
+            "file_type": label["file_type"],
+        }
+
+    def attach_shipping_label(self, label):
+        """Attach a label returned by generate_shipping_labels to a picking"""
+        if self.delivery_type != "postlogistics":
+            return super().attach_shipping_label(label)
+        self.ensure_one()
+        data = self.get_shipping_label_values(label)
+        context_attachment = self.env.context.copy()
+        # remove default_type setted for stock_picking
+        # as it would try to define default value of attachement
+        if "default_type" in context_attachment:
+            del context_attachment["default_type"]
+        return (
+            self.env["postlogistics.shipping.label"]
+            .with_context(**context_attachment)
+            .create(data)
+        )
+
+    def _set_a_default_package(self):
+        """Pickings using this module must have a package
+        If not this method put it one silently
+        """
+        # TODO: consider to depends on base_delivery_carrier_label
```

Comment:
could this migration be the opportunity to take care of this?

Reply:
This migration MUST be the opportunity to do so and will. I agreed.